File: app/src/main/java/net/daverix/ajvm/operation/L2IOperation.java
package net.daverix.ajvm.operation;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class L2IOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        long longValue = (long) currentFrame.pop();
        currentFrame.push((int) longValue);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/L2IOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class L2IOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val longValue = currentFrame.pop() as Long
        currentFrame.push(longValue.toInt())
    }
}
